src/NzbDrone.Core/DecisionEngine/Specifications/UpgradableSpecification.cs

```diff
@@ -57,9 +90,48 @@ private bool IsPreferredWordUpgradable(int currentScore, int newScore)
             return newScore > currentScore;
         }
 
+        public bool IsUpgradable(QualityProfile qualityProfile, LanguageProfile languageProfile, List<QualityModel> currentQualities, List<Language> currentLanguages, int currentScore, QualityModel newQuality, Language newLanguage, int newScore)
+        {
+            if (IsQualityUpgradable(qualityProfile, currentQualities, newQuality))
+            {
+                return true;
+            }
+
+            foreach (var quality in currentQualities)
+            {
+                if (new QualityModelComparer(qualityProfile).Compare(newQuality, quality) < 0)
+                {
+                    _logger.Debug("Existing item has better quality, skipping");
+                    return false;
+                }
+            }
+
+            if (IsLanguageUpgradable(languageProfile, currentLanguages, newLanguage))
+            {
+                return true;
+            }
+
+            foreach (var language in currentLanguages)
```

Comment:
Similarly here, could this be merged into a slightly generalized `IsLanguageUpgradable` so we only loop once?